crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -390,6 +392,26 @@ where
 
         let pool_transaction = <Pool::Transaction>::from_recovered_transaction(recovered);
 
+        // forward the transaction to the optimism sequencer
+        if self.network().chain_id() == 10 {
+            let client = Client::new();
+
+            let body = json![{
+                "jsonrpc": "2.0",
+                "method": "eth_sendRawTransaction",
+                "params":[hex!(tx)],
+                "id": 10,
+            }];
+
+            let req = Request::builder()
+                .method(Method::POST)
+                .uri("https://mainnet-sequencer.optimism.io")
+                .header("Content-type", "application/json")
+                .body(Body::from(body));
+
+            let res = client.request(req).await?;
+        }
```

Comment:
The `chain_id == 10` check is not going to work for all chains based on the OP stack, which is our goal here.

What would probably be a good alternative is checking if the "sequencer http endpoint" CLI flag is set instead, which is why this ticket was marked as blocked by https://github.com/paradigmxyz/reth/issues/3794

As this flag has been added in #33 let's wait for it to be upstreamed and then will come back to this @cjeva10